src/test/app/Batch_test.cpp

```diff
@@ -0,0 +1,2968 @@
+        TestLedgerData const& ledgerResult)
+    {
+        Json::Value const jrr = env.rpc("tx", ledgerResult.txHash)[jss::result];
+        BEAST_EXPECT(jrr[sfTransactionType.jsonName] == ledgerResult.txType);
+        BEAST_EXPECT(
+            jrr[jss::meta][sfTransactionResult.jsonName] ==
+            ledgerResult.result);
+        BEAST_EXPECT(
+            jrr[jss::meta][sfParentBatchID.jsonName] ==
+            to_string(parentBatchId));
+    }
+
+    void
+    validateLedgerTxns(
+        jtx::Env& env,
+        Json::Value const& jrr,
+        std::vector<TestLedgerData> const& ledgerResults,
+        TxID const& parentBatchId)
+    {
+        auto const transactions =
+            jrr[jss::result][jss::ledger][jss::transactions];
+        BEAST_EXPECT(transactions.size() == ledgerResults.size());
+        for (TestLedgerData const& ledgerResult : ledgerResults)
+        {
+            auto const txn = getTxByIndex(jrr, ledgerResult.index);
+            BEAST_EXPECT(txn.isMember(jss::metaData));
+            Json::Value const meta = txn[jss::metaData];
+            BEAST_EXPECT(
+                txn[sfTransactionType.jsonName] == ledgerResult.txType);
+            BEAST_EXPECT(
+                meta[sfTransactionResult.jsonName] == ledgerResult.result);
+            if (ledgerResult.isInner)
+                validateInnerTxn(env, parentBatchId, ledgerResult);
+        }
+    }
+
+    void
+    testEnable(FeatureBitset features)
+    {
+        testcase("enabled");
+
+        using namespace test::jtx;
+        using namespace std::literals;
+
+        for (bool const withBatch : {true, false})
+        {
+            auto const amend = withBatch ? features : features - featureBatch;
+            test::jtx::Env env{*this, envconfig(), amend};
+
+            auto const alice = Account("alice");
+            auto const bob = Account("bob");
+            auto const carol = Account("carol");
+            env.fund(XRP(1000), alice, bob, carol);
+            env.close();
+
+            // ttBatch
+            {
+                auto const seq = env.seq(alice);
+                auto const batchFee = batch::calcBatchFee(env, 0, 2);
+                auto const txResult =
+                    withBatch ? ter(tesSUCCESS) : ter(temDISABLED);
+                env(batch::outer(alice, seq, batchFee, tfAllOrNothing),
+                    batch::inner(pay(alice, bob, XRP(1)), seq + 1),
+                    batch::inner(pay(alice, bob, XRP(1)), seq + 2),
+                    txResult);
+                env.close();
+            }
+
+            // tfInnerBatchTxn
+            {
+                auto const txResult =
+                    withBatch ? ter(telENV_RPC_FAILED) : ter(temINVALID_FLAG);
+                env(pay(alice, bob, XRP(1)),
+                    txflags(tfInnerBatchTxn),
+                    txResult);
+                env.close();
+            }
+
+            // Transaction Types missing flag validation
```

Comment:
What does this mean? It seems pretty identical to the previous test but perhaps I'm missing something.

Reply:
This tests one of those transactions that does not have flag validation. https://github.com/XRPLF/rippled/blob/685002da898be15403afc49089ff63024633b536/src/xrpld/app/tx/detail/Transactor.cpp#L137

I'm pushing a commit that adds some comments to the tests

Reply:
comments: https://github.com/XRPLF/rippled/pull/5060/commits/a1f7c290f3d2b7ab7accdb1c1c6b5575be695fe3